presto-main/src/test/java/com/facebook/presto/operator/aggregation/groupByAggregations/AggregationTestInputImpl.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.operator.aggregation.groupByAggregations;
+
+import com.facebook.presto.operator.GroupByIdBlock;
+import com.facebook.presto.operator.aggregation.AggregationTestUtils;
+import com.facebook.presto.operator.aggregation.GroupedAccumulator;
+import com.facebook.presto.operator.aggregation.InternalAggregationFunction;
+import com.facebook.presto.spi.Page;
+import com.google.common.base.Suppliers;
+import org.testng.internal.collections.Ints;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class AggregationTestInputImpl
+        implements AggreggationTestInput
+{
+    private final Page[] pages;
+    private final InternalAggregationFunction function;
+    private int[] args;
+
+    private final int offset;
+    private final boolean isRerversed;
+
+    @SuppressWarnings("NumericCastThatLosesPrecision")
+    public AggregationTestInputImpl(InternalAggregationFunction function, Page[] pages, int offset, boolean isRerversed)
+    {
+        this.pages = pages;
+        this.function = function;
+        args = AccumulatorTestUtils.createArgs(function);
+        this.offset = offset;
+        this.isRerversed = isRerversed;
+    }
+
+    @Override
+    public GroupedAccumulator runPagesOnAccumulatorWithAssertion(long groupId, GroupedAccumulator groupedAccumulator, AggregationTestOutput expectedValue)
+    {
+        return runPagesOnAccumulatorWithAssertion(groupId, Suppliers.ofInstance(groupedAccumulator), expectedValue);
+    }
+
+    @Override
+    public GroupedAccumulator runPagesOnAccumulatorWithAssertion(long groupId, Supplier<GroupedAccumulator> accumulatorSupplier, AggregationTestOutput expectedValue)
```

Comment:
Have you seen AggregationTestUtils? It does some of what this and related classes do.

Reply:
left mostly as-is for now